storm-core/src/jvm/backtype/storm/messaging/netty/Client.java

```diff
@@ -142,6 +147,15 @@ public void run() {
                 
             }
         };
+
+        connector = new Runnable() {
+            @Override
+            public void run() {
+                if (!closing) {
+                    connect();
+                }
+            }
+        };
         
         long initialDelay = Math.min(30L * 1000, max_sleep_ms * max_retries); //max wait for 30s
```

Comment:
Either the comment or the code are incorrect.

If we want to wait a maximum of 30s, then we should use `Math.max(...)`.